### ISSUE TITLE
[no-use-before-define] False positive with optional chaining

### ISSUE THREAD

**Issue by jeroenvisser101** [typescript-eslint/typescript-eslint#1104]
**Repro**

```json
{
  "rules": {
    "@typescript-eslint/no-use-before-define": ["error"]
  }
}
```

```ts
const updatedAt = data?.updatedAt;
//                      ^^^^^^^^^
// 'updatedAt' was used before it was defined. (@typescript-eslint/no-use-before-define)
```

**Expected Result**
Would not produce any errors.

**Actual Result**
Produces an error:

```ts
// 'updatedAt' was used before it was defined. (@typescript-eslint/no-use-before-define)
```

**Additional Info**
This issue seems to only happen when the variable name is the same as the property name. For instance, the code below does not show the warning:

```ts
const timestamp = data?.updatedAt;
```

**Versions**

package | version
-- | --
@typescript-eslint/parser | `2.4.0`
TypeScript | `3.7.0-beta` (15 Oct 2019)
ESLint | `6.5.1`
node | `12.11.1`
npm | `6.11.3`
yarn | `1.19.1`

**Comment by zhongliangwang** [comment]
FWIW `eslint(no-undef)` also complains the same thing.

**Comment by bradzacher** [comment]
This is known - https://github.com/typescript-eslint/typescript-eslint/issues/1116

**Comment by peoplenarthax** [comment]
> This is known - #1116

Hej, in #1116 the do not ref to the same error, they refer to `no-undef` while here is `"@typescript-eslint/no-use-before-define"` what fails as well

**Comment by bradzacher** [comment]
That was in response to the the comment directly above, which said that `no-undef` has the error as well...

If you have an issue, please file a new one.
